page-factory-core/src/main/java/ru/sbtqa/tag/pagefactory/PageManager.java

```diff
@@ -1,43 +1,80 @@
 package ru.sbtqa.tag.pagefactory;
 
 import com.google.common.reflect.ClassPath;
-import org.apache.commons.lang3.reflect.FieldUtils;
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
-import ru.sbtqa.tag.pagefactory.annotations.PageEntry;
-import ru.sbtqa.tag.pagefactory.context.PageContext;
-import ru.sbtqa.tag.pagefactory.exceptions.PageInitializationException;
-import ru.sbtqa.tag.qautils.errors.AutotestError;
-
 import java.io.IOException;
 import java.lang.annotation.Annotation;
 import java.lang.reflect.Constructor;
+import java.lang.reflect.Field;
 import java.lang.reflect.InvocationTargetException;
 import java.net.MalformedURLException;
 import java.net.URL;
+import java.util.Collection;
+import java.util.HashMap;
 import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
 import java.util.Set;
+import org.aeonbits.owner.ConfigFactory;
+import org.apache.commons.lang3.reflect.FieldUtils;
+import org.openqa.selenium.WebDriver;
+import org.reflections.Reflections;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import ru.sbtqa.tag.pagefactory.annotations.ElementTitle;
+import ru.sbtqa.tag.pagefactory.annotations.PageEntry;
+import ru.sbtqa.tag.pagefactory.context.PageContext;
+import ru.sbtqa.tag.pagefactory.environment.Environment;
+import ru.sbtqa.tag.pagefactory.exceptions.PageInitializationException;
+import ru.sbtqa.tag.pagefactory.properties.Configuration;
+import ru.sbtqa.tag.qautils.errors.AutotestError;
+import ru.sbtqa.tag.qautils.reflect.FieldUtilsExt;
 
 public class PageManager {
 
     private static final Logger LOG = LoggerFactory.getLogger(PageManager.class);
+    private static final Map<Class<? extends Page>, Map<Field, String>> PAGES_REPOSITORY = new HashMap<>();
+    private static final Configuration PROPERTIES = ConfigFactory.create(Configuration.class);
 
     private static String pagesPackage;
+    private static PageManager pageManager;
 
     public PageManager(String pagesPackage) {
         this.pagesPackage = pagesPackage;
     }
 
+    public static PageManager getInstance() {
+        if (null == pageManager) {
+            pageManager = new PageManager(PROPERTIES.getPagesPackage());
+        }
+        return pageManager;
+    }
+
+    public static Map<Class<? extends Page>, Map<Field, String>> getPageRepository() {
+        return PAGES_REPOSITORY;
+    }
+
     /**
      * Initialize page by class
      *
      * @param page TODO
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1135)